Automattic/woocommerce-payments
# Add the WooPay Express button to the Pay for order page

Fixes #5901 

**Note: This PR ~~still needs design and~~ won't be merged until [WooPay](2104-gh-Automattic/woopay) and [WC](https://github.com/woocommerce/woocommerce/pull/37588) PRs are merged.**

#### Changes proposed in this Pull Request
1. Add the WooPay Express button to the pay-for-order page
2. Pass the pay-for-order params `order_id`, `pay_for_order`, `key`, and `billing_email` to WooPay redirect URL.

#### Screenshot ( ~~Needs design~~ )

Before             |  After
:-------------------------:|:-------------------------:
<img width="565" alt="Screen Shot 2023-03-29 at 5 15 03 PM" src="https://user-images.githubusercontent.com/56378160/228696174-0167d08d-366a-4f5b-a514-5994c1d9261d.png">  | <img width="559" alt="Screen Shot 2023-04-05 at 5 19 10 PM" src="https://user-images.githubusercontent.com/56378160/230241779-19b5c108-6282-4655-ab21-e7912a22d63d.png">



#### Testing instructions
<img width="446" alt="Screen Shot 2023-04-05 at 5 21 07 PM" src="https://user-images.githubusercontent.com/56378160/230242121-0880e42a-a848-4052-b1c4-58b2bf0993f8.png">

1. Test with [WC](https://github.com/woocommerce/woocommerce/pull/37588) PR
2. Install and activate a plugin with pay-for-order flow. ex WooCommerce-Bookings
1. Purchase the product with pay-for-order flow
1. Go to WooCommerce -> Orders
1. Find the order you just replaced and order_data metabox
1. Click on the `Customer payment page →` link next to the order status
1. Check if the WooPay Express button is append on the pay for order page shown in the above picture
1. Copy the `order_id` and the `key` from the URL
1. Click on the WooPay Express button
1. It should redirect to the WooPay checkout page
1. Confirm and compare the URL containing the params `&pay_for_order=true&order_id=YOUR_ORDER_ID&key=wc_order_YOUR_KEY&billing_email=YOUR_BILLING_EMAIL`

*

-------------------


**Post merge**
